log/log.go

```diff
@@ -0,0 +1,174 @@
+// Package log Custom logger for labstack/echo that uses sirupsen/logrus.
+package log
+
+import (
+	"io"
+
+	gommonlog "github.com/labstack/gommon/log"
+	"github.com/sirupsen/logrus"
+)
+
+// Logger defines our implementation of Labstack Echo's logger API.
+type Logger struct {
+	Entry *logrus.Entry
+}
+
+// NewLogger creates and returns a new logger.
+func NewLogger(loggerEntry *logrus.Entry) *Logger {
+	return &Logger{Entry: loggerEntry}
+}
+
+// Output returns the output writer associated with the logger.
+func (l *Logger) Output() io.Writer {
+	return l.Entry.Logger.Out
+}
+
+// SetOutput sets the output writer associated with the logger.
+func (l *Logger) SetOutput(writer io.Writer) {
+	l.Entry.Logger.SetOutput(writer)
+}
+
+// Prefix returns the prefix associated with the logger. This is a no-op.
+func (l *Logger) Prefix() string {
+	return ""
+}
+
+// SetPrefix sets the prefix associated with the logger. This is a no-op.
+func (l *Logger) SetPrefix(_ string) {
+}
+
+// Level returns the level associated with the logger. The most restrictive logging level supported by
+// labstack/echo's Logger interface is ERROR. The least restrictive logging level that it supports is
+// DEBUG.
+func (l *Logger) Level() gommonlog.Lvl {
+	if l.Entry.Level == logrus.PanicLevel {
+		return gommonlog.ERROR
+	} else if l.Entry.Logger.Level == logrus.FatalLevel {
+		return gommonlog.ERROR
+	} else if l.Entry.Logger.Level == logrus.ErrorLevel {
+		return gommonlog.ERROR
+	} else if l.Entry.Logger.Level == logrus.WarnLevel {
+		return gommonlog.WARN
+	} else if l.Entry.Logger.Level == logrus.InfoLevel {
+		return gommonlog.INFO
+	} else {
+		return gommonlog.DEBUG
+	}
+}
+
+// SetLevel sets the level associated with the logger. The most restrictive logging level supported by
+// labstack/echo's Logger interface is ERROR. The least restrictive logging level that it supports is
+// DEBUG.
+func (l *Logger) SetLevel(lvl gommonlog.Lvl) {
+	if lvl == gommonlog.ERROR {
+		l.Entry.Logger.SetLevel(logrus.ErrorLevel)
+	} else if lvl == gommonlog.WARN {
+		l.Entry.Logger.SetLevel(logrus.WarnLevel)
+	} else if lvl == gommonlog.INFO {
+		l.Entry.Logger.SetLevel(logrus.InfoLevel)
+	} else {
+		l.Entry.Logger.SetLevel(logrus.DebugLevel)
+	}
+}
+
+// SetHeader adds header fields to log messages. This is a no-op.
+func (l *Logger) SetHeader(_ string) {
+}
+
+// Print prints a log message.
+func (l *Logger) Print(args ...interface{}) {
+	l.Entry.Print(args...)
+}
+
+// Printf prints a formatted log message.
+func (l *Logger) Printf(format string, args ...interface{}) {
+	l.Entry.Printf(format, args...)
+}
+
+// Printj prints a JSON log message. This is currently a no-op.
+func (l *Logger) Printj(_ gommonlog.JSON) {
+}
```

Comment:
This is a bit weird to have a specific separate JSON logging function (rather than just an output formatter) but if we want to do this, maybe we could instantiate the logger with a separate `logrus.Logger` that has the JSON output format set? Not sure. May not matter. Anyway.

Reply:
Yeah, I thought that the interface was a little weird too. We're not using this at all, so I think I'll probably leave it unimplemented for now.